Add Sha Wujing's lance

CatsEyeXI opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Adds functionality to Sha Wujing's lance / +1 as well as distilled water.
I was not able to obtain this item in retail to gather the correct animation for use, not currently in login points
https://www.bg-wiki.com/ffxi/Sha_Wujing%27s_La._%2B1

## Steps to test these changes
Use distilled water without lance +1 equipped, get no effect message
equip lance and use another water, get regen.
use lance +1 and obtain 12 distilled waters.

Commit:
add wiring for sha wujings lance

TeoTwawki commented on scripts/items/flask_of_distilled_water.lua:
```diff
@@ -0,0 +1,26 @@
+    local mainWeapon = target:getEquipID(xi.slot.MAIN)
+
+    if
+        mainWeapon == xi.item.SHA_WUJINGS_LANCE_P1 and
```
this is a pattern we've specifically been against for as long as I can remember. as one point hundred of lines exactly like this were removed from the repo

TeoTwawki commented on sql/item_usable.sql:
```diff
@@ -2341,6 +2342,7 @@ INSERT INTO `item_usable` VALUES (21266,'gastraphetes',1,1,55,0,1,10,3600,0);
 INSERT INTO `item_usable` VALUES (21267,'annihilator',1,1,55,0,1,10,3600,0);
 INSERT INTO `item_usable` VALUES (21268,'death_penalty',1,1,55,0,1,10,3600,0);
 INSERT INTO `item_usable` VALUES (21269,'armageddon',1,1,55,0,1,10,3600,0);
+INSERT INTO `item_usable` VALUES (21868,'sha_wujings_la._+1',1,3,0,0,1,30,72000,0); -- Dispenses: Distilled Water x 12 (TODO: Verify animation)
```
> 4:51 PM]Carver: sir, do you have this item by any chance?
Image
[4:52 PM]Carver: idk the animation for this, don't think you'd approve me guessing based on past experiences 😄
[6:10 PM]TeoTwawki: I do have one. its the generic item use animation
[6:11 PM]TeoTwawki: scratch butt raise hand item get messaging

TeoTwawki replied:
😄

TeoTwawki committed:
Minor cleanup:
- Correct item dispensing animation and remove 'todo'
- Switch to modifier from slot check, because historically checking gear slots in item scripts is HEAVILY discouraged with many good reasons
- Fix a typo in an unrelated item name comment along the way.
- Add several missing for "eat raw meat" and "eat raw fish" items.

TeoTwawki commented:
I've fixed the animation ID and removed that TODO, as I also added on some missing mods related to raw meat/raw fish items while assigning ID 159 to the ability to drink distilled water
